src/rollup/BridgeCreator.sol

```diff
@@ -107,11 +124,20 @@ contract BridgeCreator is Ownable {
         ISequencerInbox.MaxTimeVariation calldata maxTimeVariation,
         BufferConfig calldata bufferConfig
     ) external returns (BridgeContracts memory) {
+        // make sure the rollup exists
+        // otherwise the rollup creator could be DOSed by frontrunning
+        if (rollup.code.length == 0) {
+            revert RollupCodeEmpty();
+        }
+
+        // use create2 salt to ensure deterministic addresses
+        bytes32 create2Salt = keccak256(msg.data);
```

Comment:
we are expecting the rollup address is already reorg safe and controlled by the sender, so we don't need to have msg.sender in the salt

Reply:
alternatively this is probably cleaner and also safe
```suggestion
        bytes32 create2Salt = keccak256(abi.encode(msg.data, msg.sender));
```